### ISSUE TITLE
setInterval should also handled as global variable.

### ISSUE THREAD

**Issue by Domvel** [palantir/tslint#4764]
### Bug Report

- __TSLint version__: `5.17.0`
- __TypeScript version__: `2.8.4` and also tested with `3.2.4`
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
setInterval(() => {
  console.log('LOL');
}, 1000);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-restricted-globals": true
  }
}
```

#### Actual behavior
The linter is still satisfied with `setInterval.`

#### Expected behavior
I expected that `setInterval` is marked as error. Because it's also a global variable (method).
All global - not own - variables should be restricted. setInterval should be called by `window.setInterval`. Only the global scope itself should be available. e.g. `window`.

Should be an error:
```ts
setInterval(() => {
  console.log('LOL');
}, 1000);
```

Should be (is already) ok:
```ts
window.setInterval(() => {
  console.log('LOL');
}, 1000);
```
Same with setTimeout. And all other properties of window.
In my case, when I hover with the mouse over `setInterval` (see reference), it shows it comes from `NodeJS.Timer`. But anyway, it's a global var.

**Comment by joechung-msft** [comment]
According to the documentation, this is by design: "This rule takes a list of strings, where each string is a global to be restricted. event, name and length are restricted by default."

**Comment by JoshuaKGoldberg** [comment]
Yup, you can add timing globals such as `setInterval` and `setTimeout` to the rule's list of flagged names:

```
{
  "rules": {
    "no-restricted-globals": [
      true,
      "event",
      "length",
      "name",
      "setInterval",
      "setTimeout"
    ]
  }
}

**Comment by Domvel** [comment]
Ok, thanks guys. My intention was to suggest this for the defaults. But maybe this rule has only non-blocking values as defaults, right? If this rule also blocks `setInterval` and `setTimeout` it will annoy some users, or not? I think: Why not? Because this rule is enabled for exactly this behaviour. Avoid using of pre-defined global variables (incl. methods). This is why I marked it as "bug". Because not expected this. But anyway, just a suggestion. :)
